### PR TITLE
Support relaxing set and binding limits

### PR DESCRIPTION
**Pull request by Guang-035**
**Motivation** [ISSUE#4165](https://github.com/KhronosGroup/glslang/issues/4165):
vkd3d-proton uses large descriptor set values to partition the set space across descriptor tables, root descriptors, root constants, and related binding categories. The previous fixed-width `layout(set)` / `layout(binding)` limits caused these shaders to fail even though the larger values are intentional for descriptor heap style workflows.

**Solution**:
- Add `--relax-set-binding-limits` / `EShMsgRelaxSetBindingLimits` as an opt-in mode for larger `layout(set)` and `layout(binding)` values.
- This updates qualifier storage and unset sentinels so large values can round-trip through parsing and mapIO, while preserving the existing limits unless the flag is enabled. The option is not gated on `GL_EXT_descriptor_heap`.

**Test**:
- Added `Test/spv.relaxSetBindingLimits.frag` to cover large GLSL `layout(set)` and `layout(binding)` values with `EShMsgRelaxSetBindingLimits`.

**NOTE**:
- When this option is enabled, the maximum accepted `layout(set)` and `layout(binding)` values are relaxed to `INT_MAX`.

### CODE REVIEW

File: glslang/MachineIndependent/ParseHelper.cpp [hunk 1]
```diff
@@ -7268,7 +7268,7 @@ void TParseContext::setLayoutQualifier(const TSourceLoc& loc, TPublicType& publi
             error(loc, "needs a literal integer", "location", "");
         return;
     } else if (id == "set") {
-        if ((unsigned int)value >= TQualifier::layoutSetEnd)
+        if (!relaxSetBindingLimits() && value >= static_cast<int>(TQualifier::layoutSetEnd))
```
Comment (dnovillo) [inline glslang/MachineIndependent/ParseHelper.cpp:7271]: This will now silently accept negative values. Guard against them? Also, please add a test that expects an error for negative values.

File: glslang/MachineIndependent/ParseHelper.cpp [hunk 2]
```diff
@@ -7280,7 +7280,7 @@ void TParseContext::setLayoutQualifier(const TSourceLoc& loc, TPublicType& publi
     } else if (id == "binding") {
         profileRequires(loc, ~EEsProfile, 420, E_GL_ARB_shading_language_420pack, "binding");
         profileRequires(loc, EEsProfile, 310, nullptr, "binding");
-        if ((unsigned int)value >= TQualifier::layoutBindingEnd)
+        if (!relaxSetBindingLimits() && value >= static_cast<int>(TQualifier::layoutBindingEnd))
```
Comment (dnovillo) [inline glslang/MachineIndependent/ParseHelper.cpp:7283]: LIkewise here.

File: glslang/MachineIndependent/iomapper.cpp [hunk 3]
```diff
@@ -61,6 +61,16 @@
...
+static int getLayoutSetLimit(bool relaxSetBindingLimits)
+{
+    return relaxSetBindingLimits ? INT_MAX : int(TQualifier::layoutSetEnd);
+}
+
+static int getLayoutBindingLimit(bool relaxSetBindingLimits)
+{
+    return relaxSetBindingLimits ? INT_MAX : int(TQualifier::layoutBindingEnd);
+}
+
```
Comment (dnovillo) [inline glslang/MachineIndependent/iomapper.cpp:73]: These are used on auto-map, right? Could you add a test that exercises this code?

**Review by dnovillo — CHANGES_REQUESTED** [review]

File: glslang/MachineIndependent/ParseHelper.cpp (hunk 2 above)
Comment (Guang-035) [reply glslang/MachineIndependent/ParseHelper.cpp:7283]: Fixed

File: glslang/MachineIndependent/iomapper.cpp (hunk 3 above)
Comment (Guang-035) [reply glslang/MachineIndependent/iomapper.cpp:73]: Done. I added a GLSL IoMap auto-map test with sampler/texture uniforms that do not specify layout(binding), so the IoMap auto-map path assigns large bindings (70000/70001) with relaxSetBindingLimits enabled.

File: glslang/MachineIndependent/ParseHelper.cpp (hunk 1 above)
Comment (Guang-035) [reply glslang/MachineIndependent/ParseHelper.cpp:7271]: Done. The shared guard at `ParseHelper.cpp:7221-7224` rejects negative layout-id values, and I added a test covering both `set = -1` and `binding = -1` under relaxSetBindingLimits.

**Review by arcady-lunarg — APPROVED** [review]
